### PR TITLE
Extract comments rendering

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
Use `ContentRenderer` for evaluating comment's markdown.  Evaluate with `ContentRenderer` when the feature flag is enabled, evaluate the old way when it's not
I followed a pattern from Joshua's pr #18754

## Related Tickets & Documents
- Related Issue #18649
- Closes [#18650](https://github.com/forem/forem/issues/18650)

## Added/updated tests?
This pr changes the implementation details, not the behavior, so the existing tests apply.

## [Forem core team only] How will this change be communicated?
- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

File: app/models/comment.rb
Comment (Ridhwana) [inline app/models/comment.rb:225]: Could this and the one below be a private method?

**Review by jaw6 — APPROVED** [review]

**Comment by mirie** [comment]
I'd like to do some light QA before we close the issue. It should be fine to merge after approvals as the feature flag is disabled.

File: app/models/comment.rb
Comment (lightalloy) [reply app/models/comment.rb:225]: :thinking: Aren't they private already? (`private` is on `:193`)

Comment (Ridhwana) [reply app/models/comment.rb:225]: Oops sorry @lightalloy. I didn't see the private.

**Review by Ridhwana — APPROVED** [review]

**Pull request merged**
